lib/macho/macho_file.rb

```diff
@@ -204,13 +202,8 @@ def dylib_id
     # @return [void]
     # @raise [ArgumentError] if `new_id` is not a String
     def dylib_id=(new_id)
-      if !new_id.is_a?(String)
-        raise ArgumentError.new("argument must be a String")
-      end
-
-      if !dylib?
-        return nil
-      end
+      raise ArgumentError.new("argument must be a String") unless new_id.is_a?(String)
```

Comment:
I guess it's more consistent with most of the other code in the project, thus I'm fine with leaving it as-is and revisiting at a later stage.